homeassistant/components/ozw/config_flow.py

```diff
@@ -1,24 +1,174 @@
 """Config flow for ozw integration."""
+import logging
+
+import voluptuous as vol
+
 from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import AbortFlow
 
+from .const import CONF_INTEGRATION_CREATED_ADDON
 from .const import DOMAIN  # pylint:disable=unused-import
 
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ADDON_DEVICE = "device"
+CONF_ADDON_NETWORK_KEY = "network_key"
+CONF_NETWORK_KEY = "network_key"
+CONF_USB_PATH = "usb_path"
+CONF_USE_ADDON = "use_addon"
 TITLE = "OpenZWave"
 
+ON_SUPERVISOR_SCHEMA = vol.Schema({vol.Optional(CONF_USE_ADDON, default=False): bool})
+
 
 class DomainConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for ozw."""
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
 
+    def __init__(self):
+        """Set up flow instance."""
+        self.addon_config = None
+        self.network_key = None
+        self.usb_path = None
+        self.use_addon = False
+        # If we install the add-on we should uninstall it on entry remove.
+        self.integration_created_addon = False
+
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
         if self._async_current_entries():
             return self.async_abort(reason="single_instance_allowed")
+
+        if not self.hass.components.hassio.is_hassio():
+            return self._async_use_mqtt_integration()
+
+        return await self.async_step_on_supervisor()
+
+    @callback
+    def _async_use_mqtt_integration(self):
+        """Handle logic when using the MQTT integration."""
         if "mqtt" not in self.hass.config.components:
             return self.async_abort(reason="mqtt_required")
+
+        return self.async_create_entry(
+            title=TITLE,
+            data={
+                CONF_USB_PATH: self.usb_path,
+                CONF_NETWORK_KEY: self.network_key,
+                CONF_USE_ADDON: self.use_addon,
+                CONF_INTEGRATION_CREATED_ADDON: self.integration_created_addon,
+            },
+        )
+
+    async def async_step_on_supervisor(self, user_input=None):
+        """Handle logic when on Supervisor host."""
```

Comment:
Comment mostly restates method name... As an outsider, I'm left wondering: so on a Supervisor host, is MQTT managed outside of the MQTT integration? Can the comment explain this?

Reply:
This is the entry point for all logic that should happen when we're running with the Supervisor.

Reply:
Does running with the Supervisor imply that we're *not* running with MQTT? Sorry, outsider eyes 👀

Reply:
No, running with the Supervisor means that we can ask the user if they want to use the add-on. Later we plan to default to use the add-on and only ask the question if the user is using advanced mode. If the user doesn't want to use the add-on the MQTT integration will be required.

Currently the MQTT integration is required regardless, since we haven't implemented the direct connection yet.